sql/parser/eval.go

```diff
@@ -319,109 +319,236 @@ type cmpArgs struct {
 	rightType reflect.Type
 }
 
-var cmpOpResultType = reflect.New(reflect.TypeOf(cmpOps).Elem().Out(0)).Elem().Interface().(DBool)
+type cmpOp struct {
+	fn     func(Datum, Datum, cmpOpCacheVal) (DBool, error)
+	initFn func(Datum, Datum) (cmpOpCacheVal, error)
+}
+
+type cmpOpCacheVal interface{}
+
+var cmpOpResultType = reflect.New(reflect.TypeOf(cmpOp{}).Field(0).Type.Out(0)).Elem().Interface().(DBool)
 
 // cmpOps contains the comparison operations indexed by operation type and
 // argument types.
-var cmpOps = map[cmpArgs]func(Datum, Datum) (DBool, error){
-	cmpArgs{EQ, stringType, stringType}: func(left Datum, right Datum) (DBool, error) {
-		return DBool(left.(DString) == right.(DString)), nil
+var cmpOps = map[cmpArgs]cmpOp{
+	cmpArgs{EQ, stringType, stringType}: {
+		fn: func(left Datum, right Datum, cache cmpOpCacheVal) (DBool, error) {
```

Comment:
Replaced the parameter name in these cases with an underscore to signify that they are not being used.